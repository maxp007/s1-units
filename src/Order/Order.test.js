import React from 'react';
import Order from './Order';
import {sortTypes} from '../utils/sortOrders';
import {fakeOrders} from '../mock/fakeOrders'
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('App component', () => {
    const wrapper = shallow(<Order order={fakeOrders[0]}/>);

    it('render fakeorder[0]', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    const wrapper1 = shallow(<Order order={fakeOrders[1]}/>);

    it('render fakeorder[1]', () => {
        expect(toJson(wrapper1)).toMatchSnapshot();
    });

    const wrapper2 = shallow(<Order order={fakeOrders[2]}/>);

    it('render fakeorder[2]', () => {
        expect(toJson(wrapper2)).toMatchSnapshot();
    });

    const wrapper_ = shallow(<Order order={{}}/>);

    it('render wrapper is empty object', () => {
        expect(toJson(wrapper_)).not.toEqual(toJson(wrapper1));
    });

});