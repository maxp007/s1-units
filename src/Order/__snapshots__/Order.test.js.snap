// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render fakeorder[0] 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`App component render fakeorder[1] 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`App component render fakeorder[2] 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Эльдорадо
    </span>
    <span>
      14 марта, чт, 2019 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Ноутбук Apple MacBook Air 13.3" (MQD32RU/A)
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Игровая приставка Sony PlayStation 4 Pro 1TB Black (CUH-7208B)
    </div>
  </div>
</div>
`;
